zoo/src/main/scala/com/intel/analytics/zoo/feature/common/FeatureLabelPreprocessing.scala

```
@@ -37,13 +37,19 @@ class FeatureLabelPreprocessing[F, L, T: ClassTag](
 
   override def apply(prev: Iterator[(F, Option[L])]): Iterator[Sample[T]] = {
     prev.map { case (feature, label ) =>
-      val featureTensor = featureStep(Iterator(feature)).next()
-      label match {
-        case Some(l) =>
-          val labelTensor = labelStep(Iterator(l)).next()
-          Sample[T](featureTensor, labelTensor)
-        case None =>
-          Sample[T](featureTensor)
+      try {
+        val featureTensor = featureStep(Iterator(feature)).next()
+        label match {
+          case Some(l) =>
+            val labelTensor = labelStep(Iterator(l)).next()
+            Sample[T](featureTensor, labelTensor)
+          case None =>
+            Sample[T](featureTensor)
+        }
+      } catch {
+        case e: Exception =>
+          println(e.getMessage)
+          null
```

If we choose to set handleInvalid as error, I think we need to throw the original exception not the NullPointerException which is confusing to the user.

---

Throw a meaningful exception

---

`null` here is the place holder for the invalid data.